src/test/test-execute.c

```diff
@@ -70,6 +71,18 @@ static void check(Manager *m, Unit *unit, int status_expected, int code_expected
         assert_se(service->main_exec_status.code == code_expected);
 }
 
+static bool is_inaccessible_available(void) {
+        if (is_reg("/run/systemd/inaccessible/reg", true) <= 0 ||
+            is_dir("/run/systemd/inaccessible/dir", true) <= 0 ||
+            is_block_node("/run/systemd/inaccessible/blk") <= 0 ||
+            is_char_node("/run/systemd/inaccessible/chr") <= 0 ||
+            is_fifo("/run/systemd/inaccessible/fifo") <= 0 ||
+            is_sock("/run/systemd/inaccessible/sock") <= 0) {
+                return false;
```

Comment:
Hmm, checking the actual type appears unnecessary. I think a simple access(F_OK) based check should be sufficient to test these, no? This 

It also allows writing this in a friendlier way:

```
FOREACH_STRING(p, "/run/systemd/inaccesible/reg", "/run/systemd/inaccessible/dir", …)
        if (access(p, F_OK) < 0)
                return false;
return true;
```

or something like that...